### ISSUE TITLE
Audit on elrond-eth-bridge, go code section, commit 6c09050006f79d441c851371e8315dcbf99a6745

### ISSUE THREAD

**Issue by iulianpascalau** [ElrondNetwork/elrond-eth-bridge#15]
1. ProcessReceivedMessage from relay.go can be split in 3 different methods (part of 3 subcomponents, each implementing MessageProcessor and each registered to the same r.messenger on the desired topics). The switch from L179 can be avoided
 2. Private key in the config.toml file
 3. Tests are unable to run due to a missing .h file. Maybe provide some info how to obtain it or better, include it here?
``` 
# github.com/karalabe/usb
../vendor/github.com/karalabe/usb/hid_enabled.go:22:10: fatal error: ./hidapi/hidapi/hidapi.h: No such file or directory
   22 | #include "./hidapi/hidapi/hidapi.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
FAIL	github.com/ElrondNetwork/elrond-eth-bridge/relay [build failed]
```
 4. Some code styling issues found like missing comments on exported elements.
 5. Maybe replace the use of "math/rand" and use "crypto/rand" instead.
 
In general, the code looks organized, well written. The use of interfaces decouples the code and helps writing unit tests.

**Comment by alexandru-calinoiu** [comment]
Thank you for your feedback.

2. Can you please suggest an alternative? My plan is to include an `config.toml.example` in the repo and instruct the user to `cp config.toml.example config.toml` and update the values with their specific config.

5. I use rand to sleep a random number of seconds before broadcasting that I've joined the network

```go
	rand.Seed(time.Now().UnixNano())
	v := rand.Intn(5)

	select {
	case <-r.timer.After(time.Duration(v) * time.Second):
		r.messenger.Broadcast(JoinTopicName, []byte(JoinTopicName))
	case <-ctx.Done():
	}
```

Is just a precation in case a lot of relayers join in the same time, I doubt it will even be the case for us as the number of relayers is capped to 10 now. 
In conclusion I don't think crypto/math is worth the overhead let me know if you think otherwise.

**Comment by ccorcoveanu** [comment]
@iulianpascalau is this issue still relevant or should we close it?

**Comment by iulianpascalau** [comment]
closing
